src/components/pages/lessons/overlay/go-pro-cta-overlay.tsx

```diff
@@ -94,69 +97,82 @@ const GoProCtaOverlay: FunctionComponent<JoinCTAProps> = ({lesson}) => {
     if (!priceId) return
     if (!formik.values.email) return
 
-    const account = first<StripeAccount>(get(viewer, 'accounts'))
-    await track('checkout: selected plan', {
-      lesson: lesson.slug,
-      priceId: priceId,
-      location: 'lesson overlay',
-    })
+    let leaveSpinningForRedirect: boolean = false
 
-    const emailRequiresAProCheck =
-      !isEmpty(formik.values.email) && formik.values.email !== viewer?.email
+    setLoaderOn(true)
 
-    if (emailRequiresAProCheck) {
-      const {hasProAccess} = await axios
-        .post(`/api/users/check-pro-status`, {
-          email: formik.values.email,
-        })
-        .then(({data}) => data)
+    try {
+      const account = first<StripeAccount>(get(viewer, 'accounts'))
+      await track('checkout: selected plan', {
+        lesson: lesson.slug,
+        priceId: priceId,
+        location: 'lesson overlay',
+      })
+
+      const emailRequiresAProCheck =
+        !isEmpty(formik.values.email) && formik.values.email !== viewer?.email
 
-      if (hasProAccess) {
-        const message = `You've already got a pro account at ${formik.values.email}. Please sign in.`
+      if (emailRequiresAProCheck) {
+        const {hasProAccess} = await axios
+          .post(`/api/users/check-pro-status`, {
+            email: formik.values.email,
+          })
+          .then(({data}) => data)
 
-        toast.error(message, {
-          duration: 6000,
-          icon: '❌',
+        if (hasProAccess) {
+          const message = `You've already got a pro account at ${formik.values.email}. Please sign in.`
+
+          toast.error(message, {
+            duration: 6000,
+            icon: '❌',
+          })
+
+          track('checkout: existing pro account found', {
+            lesson: lesson.slug,
+            email: formik.values.email,
+            location: 'lesson overlay',
+          })
+
+          // email is already associated with a pro account, return early instead
+          // of sending the user to a Stripe Checkout Session.
+          return
+        }
+      }
+
+      if (emailIsValid(formik.values.email)) {
+        await track('checkout: valid email present', {
+          priceId: priceId,
+          location: 'lesson overlay',
+        })
+        await track('checkout: redirect to stripe', {
+          lesson: lesson.slug,
+          priceId,
+          location: 'lesson overlay',
         })
 
-        track('checkout: existing pro account found', {
+        stripeCheckoutRedirect({
+          priceId,
+          email: formik.values.email,
+          stripeCustomerId: account?.stripe_customer_id,
+          authToken,
+          quantity,
+          coupon: state.context.couponToApply?.couponCode,
+        })
+
+        leaveSpinningForRedirect = true
```

Comment:
There is about half a second between this function completing and browser navigating to Stripe Checkout. Instead of flickering out of the loading state right before it navigates, this boolean keeps the button spinning as the browser navigates away. It's a small touch that gives it some extra polish.